airflow/models/dagrun.py

```diff
@@ -336,7 +336,10 @@ def find(
         if dag_ids:
             qry = qry.filter(cls.dag_id.in_(dag_ids))
         if run_id:
-            qry = qry.filter(cls.run_id == run_id)
+            if isinstance(run_id, list):
+                qry = qry.filter(cls.run_id.in_(run_id))
+            else:
+                qry = qry.filter(cls.run_id == run_id)
```

Comment:
```python
if is_container(run_id):
    ...
elif run_id is not None:
    ...
```

and we can relax the type of `run_id` to `Iterable`. (And yes the same applies to `execution_date` but that’s existing code.)